src/Components/TimePicker/TimePicker.js

```diff
@@ -75,8 +75,14 @@ const TimePicker = props => {
       setShowErr(false);
     } else if (timeValue && !open && Object.values(err).length) {
       setShowErr(true);
+    } else if ( // If dropdown is open and there's a full timeValue, close dropdown
+      open
+      && timeValue
+      && timeValue.match(/((1[0-2]|0?[1-9]):([0-5][0-9]) ?([AaPp][Mm]))/)
+    ) {
+      handleMaskClick();
     }
-  }, [timeValue, open]);
```

Comment:
Seems weird to me we don't want `open` in the dependencies array, but I'm assuming it was causing some issues?

Reply:
Yes, cause I added a new condition to close the dropdown upon selection.  If it runs when the dropdown opens too, then it would just close itself as it's passing the new logic.  This will now fire when a timeValue is updated/selected only.